src/main.rs

```diff
@@ -125,6 +125,77 @@ async fn listen_blocks(
     while let Some(_handled_message) = handle_messages.next().await {}
 }
 
+/// Takes `home_dir` and `RunArgs` to build proper IndexerConfig and returns it
+async fn construct_near_indexer_config(
+    home_dir: std::path::PathBuf,
+    args: configs::RunArgs,
+) -> near_indexer::IndexerConfig {
+    // If sync_mode is SyncFromInterruption we need to check delta and find the latest known
+    // block, otherwise we build IndexerConfig as usual
+    if let configs::SyncModeSubCommand::SyncFromInterruption(interruption_args) = args.sync_mode {
+        // If delta is 0 we just return IndexerConfig with sync_mode FromInterruption
+        // without any changes
+        if interruption_args.delta == 0 {
+            return near_indexer::IndexerConfig {
+                home_dir,
+                sync_mode: near_indexer::SyncModeEnum::FromInterruption,
+                await_for_node_synced: if args.stream_while_syncing {
+                    near_indexer::AwaitForNodeSyncedEnum::StreamWhileSyncing
+                } else {
+                    near_indexer::AwaitForNodeSyncedEnum::WaitForFullSync
+                },
```

Comment:
Extract this to a variable before the line 133, so you don't need to copy-paste this logic over and over again.